src/claimtrie.cpp

```diff
@@ -576,11 +586,28 @@ bool CClaimTrieCacheBase::ReadFromDisk(const CBlockIndex* tip)
     base->clear();
     boost::scoped_ptr<CDBIterator> pcursor(base->db->NewIterator());
 
+    LogPrintf("[0%%]..."); /* Continued */
+    int totalRecords = 0;
     for (pcursor->SeekToFirst(); pcursor->Valid(); pcursor->Next()) {
+        totalRecords++;
+    }
```

Comment:
This pre-count loop could be very expensive, as the disk cache won't be warm. How long is this loop taking compared to the lower loop?

Reply:
I was seeing 1-3 seconds in the precount loop vs ~2 minutes for the lower loop.

Reply:
Sorry, I should clarify. ~2 minutes for the whole load from disk. It was split pretty evenly between the 2 lower loops (so ~1 minute each). That's why I decided to do a precount loop: counting the total in the first lower loop would leave the operation already half done (time-wise). I couldn't see another way to estimate progress, and the added time seemed negligible next to the main operation.

Reply:
Did you want me to find a way to eliminate the precount, or is this good?